main.py

```diff
@@ -503,7 +504,15 @@ def main(cl_arguments):
     if args.do_target_task_training:
         # Train on target tasks
         pre_target_train_path = setup_target_task_training(args, target_tasks, model, strict)
-
+        # Check for previous target train checkpoints
+        task_directory = check_for_previous_checkpoints(
+            args.run_dir, target_tasks, "target_train", args.load_model
+        )
+        if len(task_directory) > 0:
+            # If there is a task to restore form, target train only on target tasks including and following that task.
+            name_to_task = collections.OrderedDict([(task.name, task) for task in target_tasks])
+            last_task_index = list(name_to_task.keys()).index(task_directory)
```

Comment:
Should be able to get rid of collections and use something like target_tasks.index(task_directory)